pkg/provider/acme/provider.go

```diff
@@ -340,6 +357,62 @@ func (p *Provider) getClient() (*lego.Client, error) {
 	return p.client, nil
 }
 
+func (p *Provider) createHTTPClient() (*http.Client, error) {
+	tlsConfig, err := p.createClientTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return &http.Client{
+		Timeout: 2 * time.Minute,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			TLSHandshakeTimeout:   30 * time.Second,
+			ResponseHeaderTimeout: 30 * time.Second,
+			TLSClientConfig:       tlsConfig,
+		},
+	}, nil
+}
+
+func (p *Provider) createClientTLSConfig() (*tls.Config, error) {
+	if len(p.CACertificates) > 0 {
+		certPool, err := lego.CreateCertPool(p.CACertificates, p.CAUseSystemCertPool)
+		if err != nil {
+			return nil, err
+		}
+
+		return &tls.Config{
+			ServerName: p.CATLSServerName,
+			RootCAs:    certPool,
+		}, nil
+	}
+
+	// Compatibility layer with the lego.
+	// https://github.com/go-acme/lego/blob/834a9089f143e3407b3f5c8b93a0e285ba231fe2/lego/client_config.go#L24-L34
+	// https://github.com/go-acme/lego/blob/834a9089f143e3407b3f5c8b93a0e285ba231fe2/lego/client_config.go#L97-L113
+
+	customCACertsPath := os.Getenv("LEGO_CA_CERTIFICATES")
+	if customCACertsPath == "" {
+		return nil, nil
+	}
+
+	useSystemCertPool, _ := strconv.ParseBool(os.Getenv("LEGO_CA_SYSTEM_CERT_POOL"))
+
+	certPool, err := lego.CreateCertPool(strings.Split(customCACertsPath, string(os.PathListSeparator)), useSystemCertPool)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tls.Config{
+		ServerName: os.Getenv("LEGO_CA_SERVER_NAME"),
+		RootCAs:    certPool,
+	}, nil
```

Comment:
Here I am a bit confused, If I am not mistaken because of https://github.com/traefik/traefik/pull/10816/files#diff-4f93ba21a09614847c01e2f916eba5dcc78e000cb9c076900dc033aa5d2bfc58R272 this code will never be called right? Maybe just removing https://github.com/traefik/traefik/pull/10816/files#diff-4f93ba21a09614847c01e2f916eba5dcc78e000cb9c076900dc033aa5d2bfc58R272 is enough, wdyt?

Reply:
> this code will never be called right?

This code will be called if you are using only env vars.
The env vars and the configuration should not be mixed.